src/authoring/hooks/use-curriculum.ts

```diff
@@ -97,6 +74,47 @@ export const useCurriculum = (auth: Auth, api: AuthoringApi) => {
     }
   }, [auth.firebaseToken, auth.gitHubToken, processHash]);
 
+  useEffect(() => {
+    const onFilesChange = (snapshot: firebase.database.DataSnapshot) => {
+      const values = snapshot.val() ?? {};
+      // Firebase does not allow certain characters in keys, so they are escaped when stored.
+      const escapedValues = Object.keys(values).reduce<IUnitFiles>((acc, key) => {
+        const escapedKey = decodeURIComponent(key);
+        acc[escapedKey] = values[key];
+        return acc;
+      }, {});
+      setFiles(escapedValues);
+    };
+
+    // prevent unnecessary fetches
+    if (branch && unit && lastUnitRef.current !== unit) {
+      lastUnitRef.current = unit;
+
+      // Setup a Firebase listener for the unit file changes so that we get real-time updates.
+      // We only do direct reads - writes go through the API.
```

Comment:
This confused me a little. It looks like there is still a read of the unit content.json file going through the API. If I'm right about that, updating this comment a little to indicate that would helpful.